lib/modules/datasource/metadata.spec.ts

```diff
@@ -256,4 +260,166 @@ describe('modules/datasource/metadata', () => {
       'https://example.com/foo/bar'
     );
   });
+
+  it('Should remove homepage and set to sourceUrl when not a link to a path in repo', () => {
+    const dep = {
+      homepage: 'https://github.com/foo/bar', // link is at repo root
+      releases: [
+        { version: '1.0.1', releaseTimestamp: '2000-01-01T12:34:56' },
+        { version: '1.0.2', releaseTimestamp: '2000-01-02T12:34:56.000Z' },
+        { version: '1.0.3', releaseTimestamp: '2000-01-03T14:34:56.000+02:00' },
+      ],
+    };
+    addMetaData(dep, MavenDatasource.id, 'foobar');
+    expect(dep).toMatchObject({
+      releases: [
+        {
+          version: '1.0.1',
+          releaseTimestamp: '2000-01-01T12:34:56.000Z',
+        },
+        {
+          version: '1.0.2',
+          releaseTimestamp: '2000-01-02T12:34:56.000Z',
+        },
+        {
+          version: '1.0.3',
+          releaseTimestamp: '2000-01-03T12:34:56.000Z',
+        },
+      ],
+      sourceUrl: 'https://github.com/foo/bar',
+    });
+  });
+
+  it('Should delete homepage if its same as sourceUrl after massage', () => {
+    const dep = {
+      homepage: 'http://www.somesource.com',
+      sourceUrl: 'http://somesource.com',
```

Comment:
i don't think we should tread those as same urls 🤔

Reply:
@viceice This logic wasn't newly added its what you get after the massage, there are tests that already rely on this.
For example : 

https://github.com/renovatebot/renovate/blob/8586f463ea0b182fe274c2d4c50a79792c860390/lib/modules/datasource/metadata.spec.ts#L176-L186

https://github.com/renovatebot/renovate/blob/8586f463ea0b182fe274c2d4c50a79792c860390/lib/modules/datasource/metadata.spec.ts#L188-L199

Reply:
strange, shouldn't those only work for github and gitab urls? we can fix that in seperate pr. THose shouldn't be treated equal.

Reply:
@viceice So expected result of this test should be : 
```
{
  homepage: "http://www.somesource.com",
  sourceUrl: "http://somesource.com",
  releases: [
    {
      version: "1.0.1",
      releaseTimestamp: "2000-01-01T12:34:56.000Z",
    },
    {
      version: "1.0.2",
      releaseTimestamp: "2000-01-02T12:34:56.000Z",
    },
    {
      version: "1.0.3",
      releaseTimestamp: "2000-01-03T12:34:56.000Z",
    },
  ],
}
```
?

Reply:
I think so

Reply:
@viceice My bad on this, I've made a check to remove homepage explicitly for Github and Gitlab. And just to check for removal if its equal to sourceUrl when its not Github or Gitlab.